Mainpage and search look ready...

mukhinzzz opened the pull request.

Commit:
created page-components and setted routing

Commit:
header with all functionality is ready

Commit:
changed project structure, added search

Commit:
+ Added pagination. In some situations it doesn't look good (see README.md for information), but it works.

+ Refresh MainPage design :)

Commit:
moved pagination component on page

Commit:
moved pagination on page

VictorDanilov commented on src/components/nonPageComponents/FavoriteIcon/FavoriteIcon.jsx:
```diff
@@ -0,0 +1,37 @@
+  function addToFavorites() {
+    localStorage.setItem(`${userName}FavName${id}`, title);
+    localStorage.setItem(`${userName}FavId${title}`, id);
+    toggleFavorites((isInFavorites = true));
```
Странная запись. Нельзя так присваивать значение сразу через равно в isInFavorites. toggleFavorites(true) надо написать

VictorDanilov commented on src/components/nonPageComponents/Header/HeaderButtons/ButtonsForGuest.jsx:
```diff
@@ -0,0 +1,17 @@
+  return (
+    <Space>
+      <Link to="/signin">
+        <Button key="2">Log In</Button>
```
key задавать не нужно. У тебя тут не динамический список.

Commit:
- Refactored Favorites:

Now "To favorites" button renders only for registered users.

Initial state for Favorites now coming from localStorage.

- Delete properties 'key' from Header Buttons

Commit:
sign up, sign in, logout features added

Commit:
favorites is ready

Commit:
history page is ready

Commit:
added chosen movie page, beautified favorites

Commit:
added fast search

Commit:
fast search edited

Commit:
Error Boundary is ready

Commit:
added context API and PropTypes functionality

Commit:
added custom hook

Commit:
added rtk query for movie card

Commit:
added missing 'key' props in lists renderings

Commit:
added contextAPI to favoritesList, beautified code

Commit:
beautified code

Commit:
fixed strange behaviour of fast search

Commit:
added redirect for unauthorized users

Commit:
sealed API key to .env

Commit:
replaced 'http' with 'https'
